### PR TITLE
Domain Redirect rename to Domain Forwarding (copies)

### PR DESCRIPTION
**Pull request by lupus2k**
Related to https://github.com/Automattic/dotcom-forge/issues/3515

## Proposed Changes

Change all text related to Domain Redirect and rename it to Domain Forwarding.

## Testing Instructions

* Edit a domain
* Open the Forwarding domain card
* Check if all the copies are good and makes sense.

## TO DO
- Add fallback to show translations mentioning Domain Redirect

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by lsl — APPROVED** [review]

File: client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx [hunk 1]
```diff
@@ -78,7 +84,7 @@ export default function DomainForwardingCard( {
 		if ( isError ) {
 			dispatch(
 				errorNotice(
-					translate( 'An error occurred while fetching your domain redirects.' ),
+					translate( 'An error occurred while fetching your domain forwarding.' ),
```
Comment (candy02058912) [inline client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx:80]: Would like to gain alignment on this: unifying that we won't be adding s to `forwarding`?
Context: My spell checker suggests that there's no such thing as `forwardings`

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8719935

Hi @lupus2k, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Domain forward updated and enabled.`
- `An error occurred while updating the domain forward.`
- `Domain forward deleted successfully.`
- `An error occurred while deleting the domain forward.`
- `An error occurred while fetching your domain forwarding.`
- `Forwarding to the same domain is not allowed.`
- `Domain Forwarding`
- `Forward your domain to another`

Thank you in advance!

File: client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx [hunk 2]
```diff
@@ -125,7 +131,7 @@ export default function DomainForwardingCard( {
 			// Disallow subdomain forwardings to the main domain, e.g. www.example.com => example.com
 			// Disallow same domain forwardings (for now, this may change in the future)
 			if ( url.hostname === domainName || url.hostname.endsWith( `.${ domainName }` ) ) {
-				setErrorMessage( translate( 'Redirects to the same domain are not allowed.' ) );
+				setErrorMessage( translate( 'Forwarding to the same domain are not allowed.' ) );
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx]:
```suggestion
				setErrorMessage( translate( 'Forwarding to the same domain is not allowed.' ) );
```

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 3]
```diff
@@ -351,8 +351,8 @@ const Settings = ( {
 
 		return (
 			<Accordion
-				title={ translate( 'Redirect Domain', { textOnly: true } ) }
-				subtitle={ translate( 'Redirect from your domain to another' ) }
+				title={ translate( 'Forwarding Domain', { textOnly: true } ) }
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/index.tsx]:
```suggestion
				title={ translate( 'Domain Forwarding', { textOnly: true } ) }
```

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 4]
```diff
@@ -351,8 +351,8 @@ const Settings = ( {
...
+				subtitle={ translate( 'Forwarding from your domain to another' ) }
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/index.tsx]:
```suggestion
				subtitle={ translate( 'Forward your domain to another' ) }
```

File: client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx [hunk 5]
```diff
@@ -48,12 +48,15 @@ export default function DomainForwardingCard( {
 	const { updateDomainForwarding } = useUpdateDomainForwardingMutation( domainName, {
 		onSuccess() {
 			dispatch(
-				successNotice( translate( 'Domain redirect updated and enabled.' ), noticeOptions )
+				successNotice( translate( 'Domain forwarding updated and enabled.' ), noticeOptions )
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx]:
```suggestion
				successNotice( translate( 'Domain forward updated and enabled.' ), noticeOptions )
```

File: client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx [hunk 6]
```diff
@@ -48,12 +48,15 @@ export default function DomainForwardingCard( {
...
 			);
 		},
 		onError() {
 			dispatch(
-				errorNotice( translate( 'An error occurred while updating the redirect.' ), noticeOptions )
+				errorNotice(
+					translate( 'An error occurred while updating the forwarding.' ),
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx]:
```suggestion
					translate( 'An error occurred while updating the domain forward.' ),
```

File: client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx [hunk 7]
```diff
@@ -63,12 +66,15 @@ export default function DomainForwardingCard( {
 		onSuccess() {
 			setTargetUrl( '' );
 			dispatch(
-				successNotice( translate( 'Domain redirect deleted successfully.' ), noticeOptions )
+				successNotice( translate( 'Domain forwarding deleted successfully.' ), noticeOptions )
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx]:
```suggestion
				successNotice( translate( 'Domain forward deleted successfully.' ), noticeOptions )
```

File: client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx [hunk 8]
```diff
@@ -63,12 +66,15 @@ export default function DomainForwardingCard( {
...
 			);
 		},
 		onError() {
 			dispatch(
-				errorNotice( translate( 'An error occurred while deleting the redirect.' ), noticeOptions )
+				errorNotice(
+					translate( 'An error occurred while deleting the forwarding.' ),
```
Comment (DustyReagan) [inline client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx]:
```suggestion
					translate( 'An error occurred while deleting the domain forward.' ),
```

**Comment by lupus2k** [comment]
I updated the copies with @DustyReagan's suggestions, Thanks 🎉

**Comment by lupus2k** [comment]
As we mentioned here (p1692297513866699/1692287812.771789-slack-CKZHG0QCR), we will wait for translations before merging the rebranding as the support docs will be pushed next week

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8719935).

**Pull request merged**